apps/els_lsp/src/els_methods.erl

```diff
@@ -184,54 +193,56 @@ shutdown(Params, State) ->
 %% exit
 %%==============================================================================
 
--spec exit(params(), state()) -> no_return().
+-spec exit(params(), els_server:state()) -> no_return().
 exit(_Params, State) ->
     Provider = els_general_provider,
-    Request = {exit, #{status => maps:get(status, State, undefined)}},
+    Request = {exit, #{status => maps:get(status, State)}},
     {response, _Response} = els_provider:handle_request(Provider, Request),
```

Comment:
I wonder - does it make sense to have the `els_provider:handle_request` abstraction? Wouldn't it be better to directly call the handler function? I feel like less indirection will make the code easier to follow